zookeeper-api/src/test/java/org/apache/helix/zookeeper/impl/client/TestRawZkClient.java

```diff
@@ -89,6 +89,65 @@ public void afterClass() {
     _zkClient.close();
   }
 
+  @Test
+  void testUnimplementedTypes() {
+    System.clearProperty("zookeeper.extendedTypesEnabled");
+    String parentPath = "/tmp";
+    String path = "/tmp/unimplemented";
+    long ttl = 1L;
+    _zkClient.deleteRecursively(parentPath);
+    try {
+      _zkClient.createPersistentWithTTL(path, true, ttl);
+    } catch (ZkException e) {
+      AssertJUnit.assertTrue(e.getCause() instanceof KeeperException.UnimplementedException);
+      return;
+    }
+    _zkClient.deleteRecursively(parentPath);
+    AssertJUnit.fail();
+  }
+
+  @Test
+  void testCreatePersistentWithTTL() {
+    System.setProperty("zookeeper.extendedTypesEnabled", "true");
+    String parentPath = "/tmp";
+    String path = "/tmp/createTTL";
+    _zkClient.deleteRecursively(parentPath);
+    AssertJUnit.assertFalse(_zkClient.exists(path));
+    long ttl = 1L;
+    _zkClient.createPersistentWithTTL(path, true, ttl);
+    AssertJUnit.assertTrue(_zkClient.exists(path));
+    _zkClient.deleteRecursively(parentPath);
+    System.clearProperty("zookeeper.extendedTypesEnabled");
+  }
+
+  @Test
+  void testCreatePersistentSequentialWithTTL() {
+    System.setProperty("zookeeper.extendedTypesEnabled", "true");
+    String parentPath = "/tmp";
+    String path = "/tmp/createSequentialTTL";
+    _zkClient.deleteRecursively(parentPath);
+    AssertJUnit.assertFalse(_zkClient.exists(path + "0000000000"));
+    long ttl = 1L;
+    _zkClient.createPersistent(parentPath);
+    _zkClient.createPersistentSequentialWithTTL(path, null, ttl);
+    AssertJUnit.assertTrue(_zkClient.exists(path + "0000000000"));
+    _zkClient.deleteRecursively(parentPath);
+    System.clearProperty("zookeeper.extendedTypesEnabled");
+  }
+
+  @Test
+  void testCreateContainer() {
+    System.setProperty("zookeeper.extendedTypesEnabled", "true");
+    String parentPath = "/tmp";
+    String path = "/tmp/createContainer";
+    _zkClient.deleteRecursively(parentPath);
+    AssertJUnit.assertFalse(_zkClient.exists(path));
+    _zkClient.createContainer(path, true);
```

Comment:
Would it be better to test more entry functions, with different parameter requirements?

Reply:
I added calls with the API that writes data to ZK as well. I think the rest of the new methods get called internally by calling these higher-level API. If this doesn't satisfy your concern, please let me know so I call them explicitly.

Reply:
+1

Reply:
@mgao0 I had just added some more tests before your comment, is your comment considering those tests as well?

Reply:
Thanks @rabashizade , the updated tests look good to me.